File: integrations/symfony-ariada/src/DependencyInjection/Configuration.php
<?php

declare(strict_types=1);

namespace Ariada\Symfony\DependencyInjection;

use Symfony\Component\Config\Definition\Builder\TreeBuilder;
use Symfony\Component\Config\Definition\ConfigurationInterface;

final class Configuration implements ConfigurationInterface
{
 public function getConfigTreeBuilder(): TreeBuilder
 {
 $treeBuilder = new TreeBuilder('ariada_symfony');
 $root = $treeBuilder->getRootNode();

 $root
 ->children()
 ->scalarNode('default_url')->defaultNull()->end()
 ->scalarNode('cli_command')->defaultValue('ariada')->end()
 ->scalarNode('output_dir')->defaultValue('%kernel.project_dir%/var/ariada-output')->end()
 ->scalarNode('browser')->defaultValue('chromium')->end()
 ->scalarNode('severity_threshold')->defaultValue('moderate')->end()
 ->integerNode('timeout_ms')->defaultValue(30000)->end()
 ->arrayNode('domains')
 ->scalarPrototype()->end()
 ->defaultValue([])
 ->end()
 ->end();

 return $treeBuilder;
 }
}

File: integrations/ruby-rails-ariada/lib/ariada/rails/configuration.rb
module Ariada
 module Rails
 class Configuration
 attr_accessor:cli_command,
:output_dir,
:browser,
:format,
:severity_threshold,
:timeout_ms,
:domains,
:targets

 def initialize
 @cli_command = "ariada"
 @output_dir = "ariada-output"
 @browser = "chromium"
 @format = "json"
 @severity_threshold = "moderate"
 @timeout_ms = 30_000
 @domains = []
 @targets = []
 end
 end
 end
end
